codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/S3UpdateEndpoint.java

```diff
@@ -104,15 +147,32 @@ private void writeInputGetter(GoWriter writer, Model model, SymbolProvider symbo
         });
     }
 
-    private void writeMiddlewareHelper(GoWriter writer) {
+    private void writeS3ControlMiddlewareHelper(GoWriter writer) {
```

Comment:
Nit these functions should call `writer.addUsedImport` for each import that is used. Mainly to prevent cases of these functions being used in some other context, or file that originally intended. Causing import failures.